src/Microsoft.OpenApi/Services/OpenApiUrlTreeNode.cs

```diff
@@ -235,5 +236,152 @@ public void AddAdditionalData(Dictionary<string, List<string>> additionalData)
                 }
             }
         }
+
+        /// <summary>
+        /// Write tree as Mermaid syntax
+        /// </summary>
+        /// <param name="writer">StreamWriter to write the Mermaid content to</param>
+        public void WriteMermaid(StreamWriter writer)
+        {
+            writer.WriteLine("graph LR");
+            foreach (var style in MermaidNodeStyles)
+            {
+                writer.WriteLine($"classDef {style.Key} fill:{style.Value.Color},stroke:#333,stroke-width:2px");
+            }
+
+            ProcessNode(this, writer);
+        }
+
+        /// <summary>
+        /// Dictionary that maps a set of HTTP methods to HTML color.  Keys are sorted, uppercased, concatenated HTTP methods.
+        /// </summary>
+        public static Dictionary<string, MermaidNodeStyle> MermaidNodeStyles = new Dictionary<string, MermaidNodeStyle>
```

Comment:
does this dictionary need to be public since it's only valuable to the internal process?
if yes, may I suggest to make it readonly? as well as to add it to the public APIs?

Reply:
I specifically made it public so that users can customize the color schemes and shapes for combinations of methods.  I don't currently support all combination of methods.  So, someone can change the supported combinations.

Reply:
then can you add it to the public APIs please? Also making it readonly so people can't affect it to null and derail the rest of your code.